moped-editor/src/views/projects/projectView/ProjectTeamTable.js

```diff
@@ -2,41 +2,74 @@ import React from "react";
 import { useQuery, useMutation } from "@apollo/client";
 
 // Material
-import { CircularProgress, TextField, Typography } from "@material-ui/core";
+import {
+  Chip,
+  CircularProgress,
+  TextField,
+  Typography,
+} from "@material-ui/core";
 import { Clear as ClearIcon } from "@material-ui/icons";
 import MaterialTable, { MTableEditRow } from "material-table";
 import Autocomplete from "@material-ui/lab/Autocomplete";
-import { filterObjectByKeys } from "../../../utils/materialTableHelpers";
+
 import typography from "../../../theme/typography";
 
 // Error Handler
 import ApolloErrorHandler from "../../../components/ApolloErrorHandler";
 
-import {
-  TEAM_QUERY,
-  ADD_PROJECT_PERSONNEL,
-  UPDATE_PROJECT_PERSONNEL,
-} from "../../../queries/project";
+import { TEAM_QUERY, UPSERT_PROJECT_PERSONNEL } from "../../../queries/project";
+
+import ProjectTeamRoleMultiselect from "./ProjectTeamRoleMultiselect";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+
+const useStyles = makeStyles(() => ({
+  roleChip: {
+    margin: ".25rem",
+  },
+}));
 
 const ProjectTeamTable = ({
   personnelState,
   setPersonnelState,
   projectId = null,
 }) => {
   const isNewProject = projectId === null;
+  const classes = useStyles();
 
   const { loading, error, data, refetch } = useQuery(TEAM_QUERY, {
     variables: { projectId },
     fetchPolicy: "no-cache",
   });
-  const [addProjectPersonnel] = useMutation(ADD_PROJECT_PERSONNEL);
-  const [updateProjectPersonnel] = useMutation(UPDATE_PROJECT_PERSONNEL);
+
+  const [upsertProjectPersonnel] = useMutation(UPSERT_PROJECT_PERSONNEL);
 
   if (loading || !data) return <CircularProgress />;
 
   // Get data from the team query payload
-  const personnel = data.moped_proj_personnel;
   const users = data.moped_users;
+  const personnel = {};
+
+  // For each personnel entry...
+  data.moped_proj_personnel.map(item => {
+    // If the item does not exist in the aggregated object
+    if (!personnel.hasOwnProperty(item.user_id)) {
+      // instantiate a new object & populate
+      personnel[`${item.user_id}`] = {};
+      personnel[`${item.user_id}`].user_id = item.user_id;
+      personnel[`${item.user_id}`].role_id = [item.role_id];
+      personnel[`${item.user_id}`].notes = item.notes;
+    } else {
+      // Aggregate role_ids, and notes.
+      personnel[`${item.user_id}`].role_id.push(item.role_id);
+      personnel[`${item.user_id}`].notes = (
+        (personnel[`${item.user_id}`].notes ?? "") +
```

Comment:
Cool to see how you handled the notes associated with each entry because I forgot how this sits in the DB. Would it make sense to create a new line for each so maybe they sit next to the matching role? Just a thought and not sure how possible that is.

Reply:
It's possible for sure. We would need to discuss it with a prodman.